_posts/2022-06-22-domain-management-for-multitenant-app-in-aws.markdown

```diff
@@ -0,0 +1,64 @@
+---
+layout: post
+title: Domain Management for Multi-tenant Application in AWS
+description: Architecting multi-tenant applications
+date: 2022-07-12
+author: Shermayne Lee
+image: /img/blog/multitenancy/multitenant.png
+tags: ['multitenant', 'route53', 'DNS', 'DomainManagement']
+---
+
+
+## Introduction
+
+![Domain Management Multitenant](/img/blog/domain-management/multitenantDNS.png)
+
+I have recently started building an application that has to manage multiple clients' domains within the same application instance. Having few clients doesn't cause any trouble with my implementation initially. When the number of clients grows, keeping track of the tasks needed to keep the domain stable is harder. Having clear segregation for each client is also challenging when a single application serves multiple clients. 
+
+You might think it has more business value for you to focus on maintaining and developing the application, but managing the domain is critical in keeping it live and accessible. 
+
+Before we dive into the solution, what is a multitenant application? Multitenant application has a single instance of an application, and its infrastructure and business logic serves multiple clients. The book [Why Multi-Tenancy is Key to Successful and Sustainable Software-as-a-Service](https://books.apple.com/us/book/why-multi-tenancy-is-key-to-successful-and/id419723802) written by Larry Aiken explains the why it is essential to have multitenancy for some applications as compared to a single tenant.
+
+What makes multitenancy challenging, and so as domain management for multitenant? Multitenancy is a challenging problem because you must consider your application's security, optimization, monitoring, and high availability. Each aspect of implementation could be a real challenge when it grows.
```

Comment:
```suggestion
Why can multitenancy be challenging? Multitenancy is a challenging problem because you must consider your application's security, optimisation, monitoring, and high availability. As your client base grows, faults may have a higher severity.
```

Proofreading suggestion.